conftest.py

```diff
@@ -17,7 +25,45 @@ def staffuser():
 
 
 @pytest.yield_fixture()
-def app(request):
+def storage_dir(monkeypatch, tmpdir, settings):
+    """Return a new temporary storage dir.
+    Settings are also modified to point to the directory."""
+    STORAGE_ROOT = settings.STORAGE_ROOT
+
+    # Get the paths to changes and their original values.
+    original_paths = (k for k in dir(settings.wrapped_object)
+                      if k.isupper() and not k.startswith('__'))
+    original_paths = (k for k in original_paths
+                      if k.endswith('_ROOT') or k.endswith('_DIR'))
```

Comment:
This seems pretty fragile, as its based on an implicit, unwritten naming convention for settings.

Why isn't it enough to replace `STORAGE_ROOT`? I thought all other paths were relative to it?